tests/integration/chroot.py

```diff
@@ -0,0 +1,298 @@
+import logging
+import json
+import time
+import os
+import subprocess
+import threading
+import tempfile
+import time
+import paramiko
+import pytest
+import sys
+import tempfile
+import lzma
+import tarfile
+import shutil
+import socket
+from contextlib import closing
+from novaclient import client
+from .sshclient import RemoteClient
+from . import util
+
+# Define global logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+BIN_DIR = os.path.join(os.path.dirname(__file__), "..", "..", "bin")
+
+class CHROOT:
+    """Handle CHROOT flavour"""
+
+    @classmethod
+    def fixture(cls, config):
+
+        logger.info("Starting CHROOT integration tests.")
+
+        # We need to validate basic information
+        # for 'RemoteClient' object first. Afterwards
+        # we can run regular validation
+        logger.info("Validation starting...")
+        ip = config.get("ip", "127.0.0.1")
+        logger.info(f"Using IP {ip} to connect to chroot.")
+        port = config.get("port", "2222")
+        logger.info(f"Using port tcp/{port} to connect to chroot.")
+
+        chroot = CHROOT(config)
+        cls.chroot = chroot
+
+        try:
+            ssh = RemoteClient(
+                host=ip,
+                port=port,
+                sshconfig=config["ssh"],
+            )
+            yield ssh
+        finally:
+            if ssh is not None:
+                ssh.disconnect()
+            if chroot is not None:
+                chroot.__del__()
+
+    @classmethod
+    def instance(cls):
+        return cls.chroot
+
+
+    def __init__(self, config):
+        """ Creating self config """
+        # Define self.config
+        self.config = config
+        # Perform basic validation
+        self._validate()
+        # Unarchive defined tar ball
+        tmp_dir = self._unarchive_image()
```

Comment:
In order to make it easier to understand what the content of this tmp_dir is, I would suggest to rename this variable and the corresponding parameters from `tmp_dir` to `rootfs`. It makes it easier for someone who already implemented chroot tests in `garden-build` since the variable there is called `rootfs` as well.

Reply:
Makes sense.